workflows/multiomics/integration/bbknn_leiden/main.nf

```diff
@@ -0,0 +1,105 @@
+nextflow.enable.dsl=2
+
+workflowDir = params.rootDir + "/workflows"
+targetDir = params.rootDir + "/target/nextflow"
+
+include { leiden } from targetDir + '/cluster/leiden/main.nf'
+include { umap } from targetDir + '/dimred/umap/main.nf'
+include { bbknn } from targetDir + '/neighbors/bbknn/main.nf'
+include { move_obsm_to_obs } from targetDir + '/metadata/move_obsm_to_obs/main.nf'
+
+include { readConfig; helpMessage; preprocessInputs; channelFromParams } from workflowDir + "/utils/WorkflowHelper.nf"
+include { setWorkflowArguments; getWorkflowArguments; passthroughMap as pmap } from workflowDir + "/utils/DataflowHelper.nf"
+
+config = readConfig("$workflowDir/multiomics/integration/bbknn_leiden/config.vsh.yaml")
+
+workflow {
+  helpMessage(config)
+
+  channelFromParams(params, config)
+    | view { "Input: $it" }
+    | run_wf
+    | view { "Output: $it" }
+}
+
+workflow run_wf {
+  take:
+  input_ch
+
+  main:
+  output_ch = input_ch
+    | preprocessInputs("config": config)
+    // split params for downstream components
+    | setWorkflowArguments(
+      bbknn: [
+        "obsm_input": "embedding",
```

Comment:
Hi @mauro-saporita, i think the `setWorkflowArguments` functionality requires some explanation. It basically is, for each component, the mapping between the name of the arguments in the config and the name of an argument in the component. 

For example: 
```
      umap: [ 
        "uns_neighbors": "uns_neighbors",
        "output": "output",
        "obsm_output": "obsm_umap"
      ],
```

Here, the `obsm_umap` argument must be present in the config, and it is passed to the `obsm_output` argument in the umap component: https://github.com/openpipelines-bio/openpipeline/blob/9d58372cae05477f54f4861ee3c1e925ef64a00a/src/dimred/umap/config.vsh.yaml#L47

So each of the argument on the rigt-hand side must be present in the workflow config. 
I noticed some errors in the `scanorama_leiden` workflow as well. I fixed them on the main branch.

Reply:
For `"obsm_input": "embedding"` for example, there is no argument `--embedding` in the config.
Also, please make sure that each element in the config is mapped to at least some other argument, otherwise, they will not do anything.

Reply:
Hi @DriesSchaumont thanks for the clarification, it was useful as OP uses a quite different approach of handling arguments and dataflow compared with common Nextflow pipelines.

I've added the requested changes and double checked with Povilas. Please let me know if everything looks fine.

I couldn't find any documentation about how naming convention for workflow's arguments so I used the same argument names from the component config.yaml.

One last point, I'm not able to run tests on my machine due to the mismatch between the requested image and the host platform on Linux/arm64/v8. It might take a bit for me to be able to fully test the workflow.